packages/sanity/src/structure/comments/plugin/inspector/CommentsInspector.tsx

```diff
@@ -92,6 +98,26 @@ function CommentsInspectorInner(props: DocumentInspectorProps) {
   }, [comments.loading, ready])
 
   const {selectedPath, setSelectedPath} = useCommentsSelectedPath()
+  const {upsellData, telemetryLogs} = useCommentsUpsell()
+
+  useEffect(() => {
+    if (mode === 'upsell') {
+      if (selectedPath?.origin === 'form') {
+        telemetryLogs.panelViewed('field_action')
+      } else if (selectedPath?.origin === 'url') {
```

Comment:
When I test this, it seems like the telemetry event won't trigger when navigating to a comment from a URL. I think this is due to the `selectedPath` state being set in the `useEffect` at line 314 when navigating from a URL, which happens after this particular effect has already executed. That is, the `selectedPath` state is not yet available when this effect runs when navigating from a URL. I think this can be solved by moving the link telemetry event inside the effect at line 314.

What do you think of moving the `panelDismissed` event to the `handleCloseInspector` function instead? This because effects is running twice in strict mode, so that way we can avoid that the dismissed event is triggered on mount. The downside of this is that we won't trigger any telemetry event when dismissing the inspector from the button in the document panel header. However, there is a `onClose` function available that can be passed to [the inspector definition](https://github.com/sanity-io/sanity/blob/next/packages/sanity/src/structure/comments/plugin/inspector/index.ts#L20) – maybe that one can be used to trigger the event?

Reply:
re url not working:
- Did a change, will check for the `commentIdParamRef.current` instead, which is available while loading the comments. It's weird it initially worked as I expected. But thanks for teh catch.

re document_action
- Those names are defined by the growth team so I think we need to stick to it, regarding changing it o onOpen I tried it but we don't have access to the upsell context to determine if we should trigger the event or not. 

re on dismiss:
-  Tried the onClose option, it also has an onOpen, but we don't have access to the upsell context to determine if we should trigger the event or not.